### PR TITLE
Add dockerfiles showing dependencies installation

### PR DESCRIPTION
**Pull request by pierreca**
# Description of the problem
Installing boost and its dependencies is difficult and it's hard for customers to figure out how to run our SDK.

# Description of the solution
Add a docker files folder showing how to install dependencies and build/run our SDK

### CODE REVIEW

File: dockerfiles/Dockerfile_Linux.build.i686_centos [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+RUN /opt/python/cp35-cp35m/bin/python -m virtualenv /usr/env3
+RUN source /usr/env3/bin/activate && pip install --upgrade pip && pip install -U setuptools wheel
+
+WORKDIR /usr/sdk
+
+# Copy code (this assumes the ./src folder contains the recursively cloned SDK repository (azure/azure-iot-sdk-python))
+COPY ./src ./src
+
+# Build for python 3
+# RUN source /usr/env3/bin/activate && sudo ./src/build_all/linux/setup.sh --python-version 3.5
```
Comment (jspaith) [inline dockerfiles/Dockerfile_Linux.build.i686_centos]: Should remove commented out line?

File: dockerfiles/Dockerfile_Linux.build.i686_centos [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+RUN /opt/python/cp35-cp35m/bin/pip3 install virtualenv
+RUN /opt/python/cp35-cp35m/bin/python -m virtualenv /usr/env3
+RUN source /usr/env3/bin/activate && pip install --upgrade pip && pip install -U setuptools wheel
+
+WORKDIR /usr/sdk
+
+# Copy code (this assumes the ./src folder contains the recursively cloned SDK repository (azure/azure-iot-sdk-python))
+COPY ./src ./src
+
+# Build for python 3
```
Comment (jspaith) [inline dockerfiles/Dockerfile_Linux.build.i686_centos]: Is there a reason we don't build Python2 from this?

File: dockerfiles/Dockerfile_Linux.build.i386 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+RUN apt-get update && apt-get install -y sudo cmake build-essential curl libcurl4-openssl-dev \
+    libssl-dev uuid-dev python-pip apt-utils python-virtualenv python3
+
+WORKDIR /usr/sdk
+
+RUN python -m virtualenv --python=python3 env3
+RUN source env3/bin/activate && pip install --upgrade pip && pip install -U setuptools wheel
+
+# Copy code
```
Comment (jspaith) [inline dockerfiles/Dockerfile_Linux.build.i386]: Recommend using more descriptive comment from some of other dockers, namely 
# Copy code (this assumes the ./src folder contains the recursively cloned SDK repository (azure/azure-iot-sdk-python))

File: dockerfiles/readme.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Dockerfiles for the Azure IoT SDK for Python
+
+The current implementation of the Python SDK is a wrapper over the C SDK and has external dependencies that cannot be installed with pip (Boost, libssl, curl...). The versions for these dependencies have to match exactly (because of how C linkers work).
```
Comment (jspaith) [inline dockerfiles/readme.md]: >wrapper over the C SDK [](start = 50, length = 22)

"wrapper over the C SDK"  -nit but I'd recommend spelling out "wrapper over the Azure IoT C SDK".

File: dockerfiles/Dockerfile_Linux.build.i686_centos (hunk 2 above)
Comment (digimaun) [reply dockerfiles/Dockerfile_Linux.build.i686_centos]: These dockerfiles are coming from the python build tool, this centOS build was a work in progress to fulfill manylinux1 requirements but never complete. I wouldn't recommend sharing this until issues are resolved.

File: dockerfiles/Dockerfile_Ubuntu16.04.run.x86_64 [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /usr/sdk
+
+RUN pip3 install --upgrade pip
+RUN pip install azure-iothub-device-client
+
+# Copy code (this assumes the ./src folder contains the code using the SDK and that the entrypoint is app.py)
+COPY ./src ./src
+
+CMD ["python3", "-u", "./src/app.py"]
```
Comment (jspaith) [inline dockerfiles/Dockerfile_Ubuntu16.04.run.x86_64]: It looks like this is for running a .py on Ubuntu when the whl has already been installed?  Does this sample add anything where we'd explicitly need it? -- if it does could we comment on it a bit more here and/or in the readme.md?

File: dockerfiles/Dockerfile_Linux.build.i686_centos (hunk 2 above)
Comment (pierreca) [reply dockerfiles/Dockerfile_Linux.build.i686_centos]: alright i'll remove this dockerfile alltogether. thanks!

File: dockerfiles/Dockerfile_Linux.build.i686_centos (hunk 1 above)
Comment (pierreca) [reply dockerfiles/Dockerfile_Linux.build.i686_centos]: removing the file.

File: dockerfiles/Dockerfile_Linux.build.i386 (hunk 3 above)
Comment (pierreca) [reply dockerfiles/Dockerfile_Linux.build.i386]: done

File: dockerfiles/Dockerfile_Ubuntu16.04.run.x86_64 (hunk 5 above)
Comment (pierreca) [reply dockerfiles/Dockerfile_Ubuntu16.04.run.x86_64]: correct. I still need to address this in the readme.

**Pull request merged**